src/pip/_internal/index.py

```diff
@@ -694,18 +711,23 @@ def find_candidates(
         self,
         project_name,   # type: str
         specifier=specifiers.SpecifierSet(),  # type: specifiers.BaseSpecifier
```

Comment:
One more comment I had is that in general I think using the pattern of having the default be `None` in the function signature and then setting the default value after checking `is None` at the beginning of the function / method is preferable. One advantage is that it composes better, because callers can pass `None` to get default behavior.